assets/tests/components/detours/diversionPage.test.tsx

```diff
@@ -933,6 +933,52 @@ describe("DiversionPage", () => {
     })
   })
 
+  test("has (only) an original route shape at the beginning", async () => {
+    const { container } = render(<DiversionPage />)
+
+    await waitFor(() => {
+      expect(originalRouteShape.get(container)).toBeVisible()
+    })
+
+    expect(originalRouteShape.diverted.getAll(container)).toHaveLength(0)
+  })
+
+  test("keeps the original route shape after the start point is added", async () => {
+    const { container } = render(<DiversionPage />)
+
+    act(() => {
+      fireEvent.click(originalRouteShape.get(container))
+    })
+
+    await waitFor(() => {
+      expect(originalRouteShape.get(container)).toBeVisible()
+    })
+  })
+
+  test("replaces the original route shape with a diverted segment after the end point is added", async () => {
+    jest.mocked(fetchFinishedDetour).mockResolvedValue(
+      finishedDetourFactory.build({
+        routeSegments: routeSegmentsFactory.build(),
```

Comment:
question(non-blocking): is there a reason to make this explicit but without parameters?

Reply:
Lol nope